base_multi_company/models/res_company_assignment.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models, tools
+
+
+class ResCompanyAssignment(models.Model):
+
+    _name = 'res.company.assignment'
+    _auto = False
+
+    name = fields.Char()
+
+    @api.model_cr
+    def init(self):
+        tools.drop_view_if_exists(self.env.cr, self._table)
```

Comment:
@pedrobaeza 

> There should be a reason why Odoo recreates them each time...

Before odoo 8, it was not possible to declare init hooks.  The creation of views was only possible by an override of the `init` method. This approach lead to serious performance issues when you update (-u all) an odoo instance with views declared on large tables.

The only advantage is that you don't have to provide a migration script if the view is modified....